### ISSUE TITLE
Report repetitive needless imports

### ISSUE THREAD

**Issue by lancelote** [intellij-rust/intellij-rust#4587]
## Environment

* **IntelliJ Rust plugin version:** 0.2.109.SNAPSHOT-192-dev
* **Rust toolchain version:** 1.38.0 (625451e37 2019-09-23) x86_64-apple-darwin
* **IDE name and version:** CLion 2019.2.3 (CL-192.6817.18)
* **Operating system:** macOS 10.15

## Problem description

Would love to see a warning for needless imports when I accidentally use the same object multiple times.

## Steps to reproduce

```rust
pub fn foo() -> &'static str { "foo" }

pub fn bar() -> &'static str { "bar" }

pub fn egg() -> &'static str { "egg" }

#[cfg(test)]
mod tests {
    use {
        crate::foo,
        crate::bar,
        super::foo
    };

    #[test]
    fn test_foo() {
        assert_eq!(foo(), "foo");
    }

    #[test]
    fn test_bar() { assert_eq!(bar(), "bar"); }

    #[test]
    fn test_egg() { assert_eq!(egg(), "egg"); }
}
```

No warnings are reported (assuming you have #4545), meanwhile the compilation fails:

```
error[E0252]: the name `foo` is defined multiple times
  --> src/lib.rs:14:9
   |
12 |         crate::foo,
   |         ---------- previous import of the value `foo` here
13 |         crate::bar,
14 |         super::foo
   |         ^^^^^^^^^^
   |         |
   |         `foo` reimported here
   |         help: remove unnecessary import
   |
   = note: `foo` must be defined only once in the value namespace of this module
```
